source/base/qprojector.cc

```diff
@@ -333,6 +333,117 @@ namespace internal
 
         return std::make_pair(final_subface_no, final_ref_case);
       }
+
+
+      Quadrature<2>
+      project_to_all_faces_or_subfaces(const ReferenceCell      &reference_cell,
+                                       const hp::QCollection<1> &quadrature,
+                                       const bool project_to_subface)
+      {
+        Assert(reference_cell == ReferenceCells::Triangle ||
+                 reference_cell == ReferenceCells::Quadrilateral,
+               ExcInternalError());
+
+        constexpr unsigned int dim = 2;
+
+        const unsigned int n_faces = reference_cell.n_faces();
+        const unsigned int subfaces_per_face =
+          project_to_subface ?
+            reference_cell.face_reference_cell(0).n_isotropic_children() :
+            1;
+
+        unsigned int n_points = 0;
+        if (quadrature.size() == 1)
+          n_points = quadrature[0].size() * n_faces;
+        else
+          {
+            AssertDimension(quadrature.size(), n_faces);
+            for (const auto &q : quadrature)
+              n_points += q.size();
+          }
+
+        // new (projected) quadrature points and weights
+        std::vector<Point<2>> q_points;
+        q_points.reserve(2 * n_points * subfaces_per_face);
+        std::vector<double> weights;
+        weights.reserve(2 * n_points * subfaces_per_face);
+
+        // loop over all faces ...
+        for (unsigned int face = 0; face < n_faces; ++face)
+          // ... and over all possible orientations ...
+          for (unsigned int orientation = 0; orientation < 2; ++orientation)
+            // ... and over all subfaces
+            for (unsigned int subface = 0; subface < subfaces_per_face;
+                 ++subface)
+              {
+                const unsigned int quadrature_index =
+                  quadrature.size() == 1 ? 0 : face;
+                const unsigned int n_q_points =
+                  quadrature[quadrature_index].size();
+
+                // handle the points
+                std::vector<Point<dim>> points(n_q_points);
+
+                // orientation 1 is the standard orientation
+                // so invert lines for orientation 0
+                if (project_to_subface)
+                  {
+                    const unsigned int local_subface =
+                      orientation == 0 ? subfaces_per_face - 1 - subface :
+                                         subface;
```

Comment:
```suggestion
                      orientation == numbers::reverse_line_orientation ? subfaces_per_face - 1 - subface :
                                         subface;
```
Here and elsewhere we need to use the new names for these values so that we can make 0 the default soon.

Reply:
I changed it where possible, as far as I understand changing the default 0 should now make no difference.